src/CodeGenerator/AutoloadGenerator.php

```diff
@@ -7,87 +7,94 @@
+        }
+    }
+
+    private function buildFromTemplate(string $templateFile, string $outputFile, array $replacements) : void
+    {
+        $template = file_get_contents($templateFile);
+        $code = strtr($template, $replacements);
+        $outputFile = $this->outputDirectory . '/' . $outputFile;
+
+        $this->writeFile($outputFile, $code);
+    }
+
+    private function generateClassmapCode(array &$classmap) : string
     {
-        $class = new ClassGenerator('Autoloader');
-        $classmapValue = new PropertyValueGenerator(
-            $classmap,
-            PropertyValueGenerator::TYPE_ARRAY_SHORT,
-            PropertyValueGenerator::OUTPUT_MULTIPLE_LINE
+        $lines = array_map(
+            function (string $class, string $file): string {
+                return var_export($class, true) . ' => ' . var_export($file, true) . ',';
+            },
+            array_keys($classmap),
+            $classmap
         );
 
-        $registerCode = 'if (!$this->registered) {'.PHP_EOL
-            . '    spl_autoload_register($this);'.PHP_EOL
-            . '    $this->registered = true;'.PHP_EOL
-            . '}'.PHP_EOL
-            . 'return $this;';
-
-        $unregisterCode = 'if ($this->registered) {'.PHP_EOL
-            . '    spl_autoload_unregister($this);'.PHP_EOL
-            . '    $this->registered = false;'.PHP_EOL
-            . '}'.PHP_EOL
-            . 'return $this;';
-
-        $loadCode = 'if (isset($this->classmap[$class])) {'.PHP_EOL
-            . '    include __DIR__ . \'/\' . $this->classmap[$class];'.PHP_EOL
-            . '}';
-
-        $class
-            ->addProperty('registered', false, PropertyGenerator::FLAG_PRIVATE)
-            ->addProperty('classmap', $classmapValue, PropertyGenerator::FLAG_PRIVATE)
-            ->addMethod('register', [], MethodGenerator::FLAG_PUBLIC, $registerCode)
-            ->addMethod('unregister', [], MethodGenerator::FLAG_PUBLIC, $unregisterCode)
-            ->addMethod('load', ['class'], MethodGenerator::FLAG_PUBLIC, $loadCode)
-            ->addMethod('__invoke', ['class'], MethodGenerator::FLAG_PUBLIC, '$this->load($class);');
-
-        $file = new FileGenerator();
-        $file
-            ->setDocBlock(new DocBlockGenerator('Generated autoloader for Zend\Di'))
-            ->setNamespace($this->namespace)
-            ->setClass($class)
-            ->setFilename($this->outputDirectory . '/Autoloader.php');
-
-        $file->write();
+        $indent = str_repeat(' ', 8);
+        return implode("\n$indent", $lines);
+    }
+
+    private function generateAutoloaderClass(array &$classmap) : void
+    {
+        $replacements = [
+            '%namespace%' => $this->namespace ? sprintf("namespace %s;\n", $this->namespace) : '',
+            '%classmap%' => $this->generateClassmapCode($classmap),
+        ];
+
+        $this->buildFromTemplate(self::CLASS_TEMPLATE, 'Autoloader.php', $replacements);
```

Comment:
In this and the following method, you can eliminate the temporary variable assignment by inlining it in the method call:

```php
$this->buildFromTemplate(self::CLASS_TEMPLATE, 'Autoloader.php', [
    '%namespace%' => $this->namespace ? sprintf("namespace %s;\n", $this->namespace) : '',
    '%classmap%' => $this->generateClassmapCode($classmap),
]);
```